src/main/java/io/openliberty/tools/intellij/util/LibertyMavenUtil.java

```diff
@@ -214,24 +216,35 @@ public static String getMavenSettingsCmd(Project project) throws LibertyExceptio
     /**
      * Get the local wrapper path for Maven that is in the project level
      *
-     * @param project liberty project
+     * @param project the liberty project containing the application project directory
+     * @param buildFile the build file specified in the application project directory
      * @return the Maven wrapper path to be executed or an exception to display
      * @throws LibertyException
      */
-    private static String getLocalMavenWrapper(Project project) throws LibertyException {
+    private static String getLocalMavenWrapper(Project project, VirtualFile buildFile) throws LibertyException {
         String mvnw = SystemInfo.isWindows ? ".\\mvnw.cmd" : "./mvnw";
-        File file = new File(project.getBasePath(), mvnw);
-        if (!file.exists()){
+        File wrapper = new File(buildFile.getParent().getPath(), mvnw);
+        File bfWrapper = null;
+        if (!wrapper.exists() || !wrapper.canExecute()) {
+            bfWrapper = wrapper;
+            wrapper = new File(project.getBasePath(), mvnw);
+        }
+        if (!wrapper.exists()){
             String translatedMessage = LocalizedResourceUtil.getMessage("maven.wrapper.does.not.exist");
             throw new LibertyException("A Maven wrapper for the project could not be found. Make sure to configure a " +
                     "valid Maven wrapper or change the build preferences for Maven inside IntelliJ Maven preferences.", translatedMessage);
         }
-        if (!file.canExecute()) {
+        if (!wrapper.canExecute()) {
             String translatedMessage = LocalizedResourceUtil.getMessage("maven.wrapper.cannot.execute");
             throw new LibertyException("Could not execute Maven wrapper because the process does not have permission to " +
                     "execute it. Consider giving executable permission for the Maven wrapper file or changing the build " +
                     "preferences for Maven inside IntelliJ Maven preferences.", translatedMessage);
         }
+        if (bfWrapper != null) {
+            LOGGER.warn("The wrapper built from the build file is invalid, using wrapper from project directory. exists="
+                    + bfWrapper.exists() + " canExecute=" + bfWrapper.canExecute());
```

Comment:
```suggestion
            LOGGER.warn("A wrapper in the build file directory is either invalid or missing. Using wrapper from project base directory. Build file wrapper values: exists="
                    + bfWrapper.exists() + " canExecute=" + bfWrapper.canExecute());
```

Reply:
Updated this message in the alternate branch where this code is stored for future consideration.